AppCenterIdentity/AppCenterIdentity/MSIdentity.m

```diff
@@ -141,9 +141,9 @@ + (void)handleUrlResponse:(NSURL *)url {
 }
 
 + (void)login {
-
-  // TODO protect with canBeUsed.
-  [[MSIdentity sharedInstance] login];
+  if ([[MSIdentity sharedInstance] canBeUsed]) {
```

Comment:
Consider using `@synchronized` block in static method. You might want to remove another block in instance method.

Reply:
Agree, fixed. Thank you!